pkg/controller/performanceprofile/components/machineconfig/machineconfig_test.go

```diff
@@ -167,6 +177,28 @@ var _ = Describe("Machine Config", func() {
 		Expect(manifest).To(ContainSubstring(expectedBootArgumentsWithoutIso))
 	})
 
+	It("should generate yaml with expected parameters and additional kernel arguments", func() {
+		profile := testutils.NewPerformanceProfile("test")
+		profile.Spec.AdditionalKernelArgs = append(profile.Spec.AdditionalKernelArgs,
+			"nmi_watchdog=0", "audit=0",
+			"mce=off",
+			"processor.max_cstate=1",
+			"idle=poll",
+			"intel_idle.max_cstate=0")
+		mc, err := New(testAssetsDir, profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		y, err := yaml.Marshal(mc)
+		Expect(err).ToNot(HaveOccurred())
+
+		manifest := string(y)
+
+		labelKey, labelValue := components.GetFirstKeyAndValue(profile.Spec.MachineConfigLabel)
+		Expect(manifest).To(ContainSubstring(fmt.Sprintf("%s: %s", labelKey, labelValue)))
+		Expect(manifest).To(ContainSubstring(expectedSystemdUnits))
+		Expect(manifest).To(ContainSubstring(expectedBootArgumentsWithoutAdditionalKerenlArgs))
```

Comment:
I think we can add another Expect here for the AdditionalKernelArgs?

Reply:
ah, i have a mistake in the name. should be `expectedBootArgumentsWithAdditionalKerenlArgs`  (not without)